src/cli/search.rs

```diff
@@ -33,15 +33,27 @@ impl Command for Search {
     fn signature(&self) -> Signature {
         Signature::build("search")
             .required("index", SyntaxShape::String, "the index name")
-            .required(
+            .optional(
```

Comment:
If vector is a named param then should query be the same?

Reply:
I mainly left it as unnamed for the case where we are just doing a text query. Having to do
```
search indexName --query stringQuery
```
Feels a but redundant to me

Reply:
In SDK land the query is optional, and we send a match-none query if the user doesn't specify one which is something that maybe we could do and leads into the other comment I'm about to make.